stats/opentelemetry/internal/grpc_trace_bin_propagator.go

```diff
@@ -22,111 +22,117 @@ import (
+// If the carrier is a CustomCarrier, trace data is directly injected in a
+// binary format using the 'grpc-trace-bin' header (fast path). Otherwise,
+// the trace data is base64 encoded and injected using the same header in
+// text format (slow path).
+func (p GRPCTraceBinPropagator) Inject(ctx context.Context, carrier otelpropagation.TextMapCarrier) {
+	span := oteltrace.SpanFromContext(ctx)
 	if !span.SpanContext().IsValid() {
 		return
 	}
 
-	binaryData := Binary(span.SpanContext())
-	if binaryData == nil {
+	bd := Binary(span.SpanContext())
+	if bd == nil {
 		return
 	}
 
-	if customCarrier, ok := carrier.(otelinternaltracing.CustomCarrier); ok {
-		customCarrier.SetBinary(binaryData) // fast path: set the binary data without encoding
-	} else {
-		carrier.Set(otelinternaltracing.GRPCTraceBinHeaderKey, base64.StdEncoding.EncodeToString(binaryData)) // slow path: set the binary data with encoding
+	if cc, ok := carrier.(internaltracing.CustomCarrier); ok {
+		cc.SetBinary(bd)
+		return
 	}
+	carrier.Set(internaltracing.GRPCTraceBinHeaderKey, base64.StdEncoding.EncodeToString(bd))
 }
 
-// Extract reads cross-cutting concerns from the carrier into a Context.
+// Extract reads OpenTelemetry trace context information from the carrier into a
+// Context.
 //
-// If carrier is carrier.CustomCarrier then GetBinary (fast path) is used,
-// otherwise Get (slow path) with decoding is used.
-func (p GRPCTraceBinPropagator) Extract(ctx context.Context, carrier propagation.TextMapCarrier) context.Context {
-	var binaryData []byte
-
-	if customCarrier, ok := carrier.(otelinternaltracing.CustomCarrier); ok {
-		binaryData, _ = customCarrier.GetBinary()
+// If the carrier is a CustomCarrier, trace data is read directly in a binary
+// format from the 'grpc-trace-bin' header (fast path). Otherwise, the trace
+// data is base64 decoded from the same header in text format (slow path).
+func (p GRPCTraceBinPropagator) Extract(ctx context.Context, carrier otelpropagation.TextMapCarrier) context.Context {
+	var bd []byte
+
+	if cc, ok := carrier.(internaltracing.CustomCarrier); ok {
+		bd = cc.GetBinary()
 	} else {
-		binaryData, _ = base64.StdEncoding.DecodeString(carrier.Get(otelinternaltracing.GRPCTraceBinHeaderKey))
+		bd, _ = base64.StdEncoding.DecodeString(carrier.Get(internaltracing.GRPCTraceBinHeaderKey))
 	}
-	if binaryData == nil {
+	if bd == nil {
 		return ctx
 	}
 
-	spanContext, ok := FromBinary([]byte(binaryData))
+	spanContext, ok := FromBinary([]byte(bd))
 	if !ok {
 		return ctx
 	}
 
-	return trace.ContextWithRemoteSpanContext(ctx, spanContext)
+	return oteltrace.ContextWithRemoteSpanContext(ctx, spanContext)
 }
 
-// Fields returns the keys whose values are set with Inject.
-//
-// GRPCTraceBinPropagator will only have `grpc-trace-bin` field.
+// Fields always returns a slice containing only `grpc-trace-bin` header key
+// because the GRPCTraceBinPropagator only uses the 'grpc-trace-bin' header for
+// propagating trace context.
```

Comment:
Please rewrite this comment to say what Fields() returns from a conceptual point of view. What does Fields mean? Fields of what? Why is it returning a key? Or rename the function to be semantically closer to the intended idea of what this represents.

Reply:
"Basically the propagation.TextMapCarrier interface which GRPCTraceBinPropagator implements has Keys()" your reply to another comment - should we call this Keys() instead?

Reply:
GRPCTraceBinPropagator is implementing https://github.com/open-telemetry/opentelemetry-go/blob/main/propagation/propagation.go#L82

// Fields returns the keys whose values are set with Inject.
Fields() []string

Reply:
Included // Fields returns the keys whose values are set with Inject. as well in docstring

Reply:
Oh I see. That makes sense.